doc/map.md

```diff
@@ -11,17 +11,23 @@ You can pass as many arguments as you wish to the function and
 the function can return as many values as it wants.
 
 ```lua
-   local ipc = require 'libipc'
-   local m = ipc.map(2, function(a, b, c)
-      return math.sqrt(a*a + b*b + c*c), "hi"
-   end, 1, 2, 3)
-   local p1,s1,p2,s2 = m:join()
-   print(p1) -- will print 3.7416573867739
-   print(s1) -- will print "hi"
-   print(p2) -- will print 3.7416573867739
-   print(s2) -- will print "hi"
+local ipc = require 'libipc'
+local m = ipc.map(2, function(a, b, c, threadId)
+   -- the last argument passed to the function is the ID of the thread
+   assert((threadId == 1) or (threadId == 2))
+   return math.sqrt(a*a + b*b + c*c), "hi"
+end, 1, 2, 3)
+local p1,s1,p2,s2 = m:join()
+print(p1) -- will print 3.7416573867739
+print(s1) -- will print "hi"
+print(p2) -- will print 3.7416573867739
+print(s2) -- will print "hi"
 ```
 
+Note how an additional hidden argument is passed to the mapped function: `threadId`.
+This `threadId` is a number identifying the thread.
+It is always passed as the last argument to the function.
```

Comment:
Add a note saying `threadId` starts at 1. I know it's the default in lua, but always a good reminder when dealing with low-level stuff.